sessions/recap-project-6-NEW/activity-planner/user-stories/2_activity-details.md

```diff
@@ -23,13 +23,44 @@ Group Todo: Add wireframes
   - Activity categories
   - Activity area
   - Activity country
-  - Creation date
-  - Last update date
 - The detailed view includes a "back" link, allowing users to easily navigate back to the list of activities.
 - The page implements proper error handling for non-existent activity IDs.
 - The page shows a loading state while fetching activity details.
 
 ## Tasks
 
 - [ ] Create feature branch `feature/activity-details`
-- [ ] Group Todo: Add tasks
\ No newline at end of file
+- [ ] Create API endpoint for fetching single activity by ID
+  - [ ] Implement MongoDB findById query
+  - [ ] Add proper error handling for invalid IDs
+  - [ ] Add validation for required fields
+- [ ] Create frontend components
+  - [ ] Create `ActivityDetails` page component
+  - [ ] Create `ActivityInfo` component for displaying activity data
```

Comment:
I think we should not show the pattern of a page component where we only call another component. Instead, the students can write the ActivityInfo code directly in the Details page component.